src/main/java/core/basesyntax/service/impl/FruitTransactionMapper.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.enums.Operation;
+import core.basesyntax.service.Mapper;
+
+public class FruitTransactionMapper implements Mapper<FruitTransaction, String> {
+    private static final String DELIMITER = ",";
+    private static int OPERATION_INDEX = 0;
+    private static int FRUIT_INDEX = 1;
+    private static int QUANTITY_INDEX = 2;
+
+    @Override
+    public FruitTransaction toObject(String string) {
+        String[] split = string.split(DELIMITER);
+        validateSplitLength(split, string);
+        Operation operation = Operation.fromCode(split[OPERATION_INDEX]);
+        String fruit = split[FRUIT_INDEX];
+        int quantity = parseQuantity(split[QUANTITY_INDEX]);
+        return new FruitTransaction(operation, fruit, quantity);
+    }
+
+    private void validateSplitLength(String[] split, String string) {
+        if (split.length != 3) {
+            throw new IllegalArgumentException("Input string must contain "
+                    + "exactly three elements separated by the delimiter '"
+                    + DELIMITER + "', but found " + split.length + ": [" + string + "]");
+        }
+    }
+
+    private int parseQuantity(String quantityString) {
+        int quantity = Integer.parseInt(quantityString);
```

Comment:
When parsing the quantity, a `NumberFormatException` could be thrown if the string cannot be parsed as an integer. You should catch this exception and throw an `IllegalArgumentException` with an informative message indicating that the quantity is not a valid integer.